uber/site_sections/preregistration.py

```diff
@@ -109,6 +109,14 @@ def form(self, session, message='', edit_id=None, **params):
             params.setdefault('badges', group.badges)
         else:
             attendee = session.attendee(params, ignore_csrf=True, restricted=True)
+            if attendee.found_how:
+                code = session.query(PromoCode).filter(PromoCode.code == attendee.found_how).first()
```

Comment:
Probably want an `if code is not None:` check after this, right? If no code is found then `code.price` will raise an exception, right?